pkg/elb/ingress.go

```diff
@@ -0,0 +1,78 @@
+package elb
+
+import (
+	"context"
+
+	v1 "k8s.io/api/networking/v1"
+	"k8s.io/api/networking/v1beta1"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Ingress struct {
+	v1betaIngress v1beta1.Ingress
+	v1Ingress     v1.Ingress
+}
+
+func (i *Ingress) Delete(kubernetesCS kubernetes.Interface) error {
+	if i.v1Ingress.ObjectMeta.Name != "" {
+		return kubernetesCS.NetworkingV1().Ingresses(i.v1Ingress.Namespace).Delete(context.TODO(), i.v1Ingress.Name, metav1.DeleteOptions{})
+	}
+	return kubernetesCS.NetworkingV1beta1().Ingresses(i.v1betaIngress.Namespace).Delete(context.TODO(), i.v1betaIngress.Name, metav1.DeleteOptions{})
+}
+
+func (i *Ingress) GetMetadata() metav1.ObjectMeta {
+	if i.v1Ingress.ObjectMeta.Name != "" {
+		return i.v1Ingress.ObjectMeta
+	}
+	return i.v1betaIngress.ObjectMeta
+}
+
+func (i *Ingress) GetLoadBalancersHosts() []string {
+	var hostNames []string
```

Comment:
this could be two separate structs, both implementing a `Ingress` interface for the 3 methods, but it didn't feel like it was worth it, the `If` switching isn't *too* ugly

Reply:
actually I am going to do this, its annoying me